### PR TITLE
Update JITServer Dockerfile

### PR DESCRIPTION
**Pull request by harryyu1994**
Use multi-stage builds to minimize the size of the final image.
Once the JDK is built, we will throw away all the build time dependencies and only keeping the runtime dependencies.
[skip ci]

Signed-off-by: Harry Yu <harryyu1994@gmail.com>

### CODE REVIEW

**Comment by harryyu1994** [comment]
Reduced JITServer Docker image size from 6.32 GB down to 1.99 GB.
The openj9 base docker image is 1.75GB.

**Comment by harryyu1994** [comment]
@mpirvu Hi Marius, could you please review this one?

File: buildenv/docker/jdk8/x86_64/ubuntu16/jitserver/build/Dockerfile [hunk 1]
```diff
@@ -32,37 +32,17 @@
...
 ARG openj9_repo=https://github.com/eclipse/openj9.git
 ARG omr_repo=https://github.com/eclipse/openj9-omr.git
 
 ENV JITSERVER_SUPPORT=1
 
 RUN apt-get update \
- && apt-get install -qq -y --no-install-recommends \
-    automake \
-    curl \
+  && apt-get install -qq -y --no-install-recommends \
```
[the remark this answers is not in the document]
Comment (harryyu1994) [reply buildenv/docker/jdk8/x86_64/ubuntu16/jitserver/build/Dockerfile]: We no longer needs the packages omitted and the base openj9 layer already uses gcc 7.

**Comment by harryyu1994** [comment]
Ready for review.

**Comment by mpirvu** [comment]
No changes to the source code, so testing will not reveal anything

**Pull request merged**